src/sentry/snuba/discover.py

```diff
@@ -788,6 +790,90 @@ def timeseries_query(selected_columns, query, params, rollup, reference_event=No
     return SnubaTSResult({"data": result}, snuba_filter.start, snuba_filter.end, rollup)
 
 
+def top_events_timeseries(
+    timeseries_columns,
+    selected_columns,
+    user_query,
+    params,
+    yaxis,
+    rollup,
+    limit,
+    organization,
+    referrer=None,
+):
+    orderby = ["-{}".format(axis) for axis in yaxis]
+    top_events = query(
+        selected_columns + yaxis,
+        query=user_query,
+        params=params,
+        orderby=orderby,
+        limit=limit,
+        referrer=referrer,
+    )
+
+    snuba_filter, translated_columns = get_timeseries_snuba_filter(
+        timeseries_columns + selected_columns, user_query, params, rollup
+    )
+
+    for field in selected_columns:
+        # project is handled by filter_keys already
+        if field in ["project", "project.id"]:
+            continue
+        values = list({event.get(field) for event in top_events["data"] if field in event})
+        if values and all(values):
+            snuba_filter.conditions.append([resolve_column(field), "IN", values])
+
+    result = raw_query(
+        aggregations=snuba_filter.aggregations,
+        conditions=snuba_filter.conditions,
+        filter_keys=snuba_filter.filter_keys,
+        start=snuba_filter.start,
+        end=snuba_filter.end,
+        rollup=rollup,
+        orderby="time",
+        groupby=["time"] + snuba_filter.groupby,
+        dataset=Dataset.Discover,
+        limit=10000,
+        referrer=referrer,
+    )
+
+    result = transform_results(result, translated_columns, snuba_filter)
+    issues = {}
+    if "issue" in selected_columns:
+        issues = Group.issues_mapping(
+            set([event["issue.id"] for event in top_events["data"]]),
+            params["project_id"],
+            organization,
+        )
+
+    translated_groupby = [translated_columns.get(field, field) for field in snuba_filter.groupby]
+    if "project.id" in translated_groupby:
+        translated_groupby.remove("project.id")
+        translated_groupby.append("project")
```

Comment:
I feel like there's probably a better way to do this, but I'm having trouble seeing one 🤔

Reply:
Add "project.id": "project" to the translated_columns? This seem fine.

Reply:
ooh I like that more :+1: